CMakeLists.txt

```diff
@@ -65,6 +65,13 @@ add_compile_options(-Wall)
 add_compile_options(-Werror)
 # add_compile_options(-Wextra) #TODO: enable it
 
+if (ENABLE_COVERAGE AND CMAKE_COMPILER_IS_GNUCXX)
+    include(CodeCoverage)
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${GCC_COVERAGE_COMPILE_FLAGS}")
```

Comment:
how about using `add_compile_option` here?

Reply:
Love the idea in theory, but it treated the whole expression as a single command line argument and the build failed.